core/keychain/keychain.go

```diff
@@ -366,3 +370,24 @@ func (kc *keychain) retrieveKeyPair() (privKey []byte, mnemonic string, err erro
 
 	return privKey, mnemonic, nil
 }
+
+func (kc *keychain) GetManagedThreadKey(threadKeyName string) (thread.Key, error) {
+	size := 32
+
+	privBytes, _, err := kc.retrieveKeyPair()
+	if err != nil {
+		return thread.Key{}, err
+	}
+
+	num := pbkdf2.Key(privBytes, []byte("threadID"+threadKeyName), 256, size, sha512.New)
```

Comment:
Just nitpick but this should be `"ThreadKey" + threadKeyName`

Reply:
@dmerrill6 😸 

https://github.com/FleekHQ/space-daemon/pull/191/commits/5a0a95a44edc6a31d21851f63ecd2ed0a68c5f5c#diff-a509c084f1fb203e2d90b6bd7e71b528R382